portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2729,13 +2729,16 @@ private String _buildEntryURL(DLFileVersion fileVersion)
 		String portletId = PortletProviderUtil.getPortletId(
 			FileEntry.class.getName(), PortletProvider.Action.MANAGE);
 
-		String entryURL = PortalUtil.getControlPanelFullURL(
+		String redirectURL = PortalUtil.getControlPanelFullURL(
 			fileVersion.getGroupId(), portletId, null);
 
 		String namespace = PortalUtil.getPortletNamespace(portletId);
 
-		entryURL = HttpUtil.addParameter(
-			entryURL, namespace + "mvcRenderCommandName",
+		redirectURL = HttpUtil.addParameter(
+			redirectURL, namespace + "folderId", fileVersion.getFolderId());
+
+		String entryURL = HttpUtil.addParameter(
```

Comment:
This reuse of the redirectURL is a bit confusing. Even if both the entry and redirect URL share some common parameters, I think it is better to keep the separate. Can you create a new private "_getRedirectURL()" method and use that in _buildEntryURL?

Even if there's a bit of duplication, I think it is much clearer to keep them separated.

Reply:
Perfect!